lib/IRGen/IRGenModule.cpp

```diff
@@ -927,9 +929,9 @@ void IRGenModule::emitAutolinkInfo() {
                                        }),
                         AutolinkEntries.end());
 
-  if (TargetInfo.OutputObjectFormat == llvm::Triple::COFF ||
-      TargetInfo.OutputObjectFormat == llvm::Triple::MachO ||
-      Triple.isPS4()) {
+  if ((TargetInfo.OutputObjectFormat == llvm::Triple::COFF &&
+       !Triple.isOSCygMing()) ||
```

Comment:
I think LLVM should not emit '-lFoo' into `.drectve` since ld can not recognize it. (in case of Cygwin/MinGW)
Below is the output before patch.
```
$ ./swiftc -c Hello.swift

$ objdump -s -h --section=.drectve Hello.o

Hello.o:     file format pe-x86-64

Sections:
Idx Name          Size      VMA               LMA               File off  Algn
  6 .drectve      00000025  0000000000000000  0000000000000000  00000314  2**0
                  CONTENTS, READONLY, DEBUGGING, EXCLUDE
Contents of section .drectve:
 0000 202d6c73 77696674 436f7265 202d6c73   -lswiftCore -ls
 0010 77696674 53776966 744f6e6f 6e655375  wiftSwiftOnoneSu
 0020 70706f72 74                          pport

$ ./swiftc Hello.swift
Warning: .drectve `-lswiftCore ' unrecognized
Warning: corrupt .drectve at end of def file
/tmp/Hello-a57504.o:(.text+0x5f): undefined reference to `_TTSgq5P____TFs27_allocateUninitializedArrayurFBwTGSax_Bp_'
collect2: error: ld returned 1 exit status
clang-3.7: error: linker (via gcc) command failed with exit code 1 (use -v to see invocation)
<unknown>:0: error: link command failed with exit code 1 (use -v to see invocation)

$ ./swiftc Hello.swift -lswiftSwiftOnoneSupport
Warning: .drectve `-lswiftCore ' unrecognized
Warning: corrupt .drectve at end of def file

```